frontend/src/scenes/cohorts/cohortLogic.tsx

```diff
@@ -104,27 +104,28 @@ export const cohortLogic = kea<cohortLogicType>({
     listeners: ({ actions, values, key }) => ({
         saveCohort: async ({ cohortParams, filterParams }, breakpoint) => {
             let cohort = { ...values.cohort, ...cohortParams } as CohortType
+            const isNewCohort = typeof cohort.id !== 'number'
             const cohortFormData = new FormData()
 
             for (const [itemKey, value] of Object.entries(cohort as CohortType)) {
                 if (itemKey === 'groups') {
                     if (cohort.is_static) {
-                        if (!cohort.csv && cohort.id === 'new') {
+                        if (!cohort.csv && isNewCohort) {
                             actions.setSubmitted(true)
-                            return
+                            continue
```

Comment:
Well these checks here are intended to make sure only valid payloads are submitted (e.g. you need to upload a CSV if you're using a static cohort), so I think something akin to `return` is needed. If the payload is invalid, we shouldn't submit the API request.